projects/core/src/cart/facade/cart.service.ts

```diff
@@ -1,34 +1,54 @@
 import { Injectable } from '@angular/core';
-
 import { select, Store } from '@ngrx/store';
-import { Observable, combineLatest } from 'rxjs';
-import { filter } from 'rxjs/operators';
-
-import { AuthService, UserToken } from '../../auth/index';
-
-import * as fromAction from '../store/actions';
-import * as fromSelector from '../store/selectors';
-import { ANONYMOUS_USERID, CartDataService } from './cart-data.service';
-import { StateWithCart } from '../store/cart-state';
+import { asyncScheduler, combineLatest, Observable } from 'rxjs';
+import { debounceTime, filter, map, tap } from 'rxjs/operators';
+import { AuthService } from '../../auth/index';
 import { Cart } from '../../model/cart.model';
 import { OrderEntry } from '../../model/order.model';
 import { BaseSiteService } from '../../site-context/index';
+import * as fromAction from '../store/actions';
+import { StateWithCart } from '../store/cart-state';
+import * as fromSelector from '../store/selectors';
+import { ANONYMOUS_USERID, CartDataService } from './cart-data.service';
 
 @Injectable()
 export class CartService {
-  private callback: Function;
+  prevCartUserId = null;
 
   constructor(
     protected store: Store<StateWithCart>,
     protected cartData: CartDataService,
     protected authService: AuthService,
     protected baseSiteService: BaseSiteService
-  ) {
-    this.init();
-  }
+  ) {}
 
   getActive(): Observable<Cart> {
-    return this.store.pipe(select(fromSelector.getCartContent));
+    return combineLatest(
+      this.store.select(fromSelector.getCartContent),
+      this.store.select(fromSelector.getLoading),
+      this.authService.getUserToken()
+    ).pipe(
+      debounceTime(1, asyncScheduler),
+      tap(([cart, loading, userToken]) => {
+        if (
+          this.prevCartUserId !== null &&
+          this.prevCartUserId !== userToken.userId &&
+          typeof userToken.userId !== 'undefined' &&
```

Comment:
Why we check for null in line 34 and for undefined here?

Reply:
After 10 minutes reading and thinking, I understand these conditions. Can we make them more readable?

Reply:
I added some comments.